src/ngraph/distributed/mlsl.hpp

```diff
@@ -83,10 +83,23 @@ namespace ngraph
                     throw std::runtime_error("AllReduce op supports only f32 and f64 types");
                 }
 
+                auto mlsl_reduce_type = MLSL::RT_SUM;
```

Comment:
Use `switch` and then you won't need the throw because the compiler will know you handled all the cases.

Reply:
You mean switch without `default`?

Reply:
switch doesn't work easily. Can I use `if`? element_type is also using `if`